core/builder/src/main/java/io/quarkus/builder/BuildContext.java

```diff
@@ -24,12 +24,14 @@
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public final class BuildContext {
+    private final ClassLoader classLoader;
     private final StepInfo stepInfo;
     private final Execution execution;
     private final AtomicInteger dependencies;
     private volatile boolean running;
 
-    BuildContext(final StepInfo stepInfo, final Execution execution) {
+    BuildContext(ClassLoader classLoader, final StepInfo stepInfo, final Execution execution) {
+        this.classLoader = classLoader;
```

Comment:
Better add an assertion here to make sure this is never null, or weird errors might appear